### ISSUE TITLE
Reductions have difficultly with non-builtin identity values

### ISSUE THREAD

**Issue by izaid** [libdynd/libdynd#968]
Reductions in DyND currently work with an `identity` keyword argument. If an `identity` value is passed in, this value is assigned to the result. If no` identity` value is passed in, then the reduction assigns the first element of the input array to be the identity.

While this works well in the case of simple operations like arithmetic, this fails when we have more complicated reductions like serialization. In that case, the signature is `(Dims... * Scalar) -> bytes`, and we can't assign values to bytes like that.

So we need a better way to handle identities and/or the initialization step of a reduction.

**Comment by mwiebe** [comment]
For serialization, zero-sized bytes makes sense as an identity.

**Comment by izaid** [comment]
Yeah, I agree with that. Except we are already initializing the output to be zero-sized bytes to begin with (the default `bytes` constructor). So this is redundant. Maybe we should have a way to specific a `do nothing` reduction initialization.
